hal/halx86/include/hal.h

```diff
@@ -36,9 +36,10 @@
 #include <ndk/kefuncs.h>
 #include <ndk/rtlfuncs.h>
 
-/* For MSVC, this is required before using DATA_SEG (used in pcidata) */
+/* Required before using CODE_SEG & DATA_SEG */
 #ifdef _MSC_VER
-# pragma section("INIT", read,execute,discard)
+# pragma section("INIT", read, execute, discard)
+# pragma section("INITDATA", read, write, discard)
```

Comment:
Regarding DATA_SEG, a similar fix appears to be necessary for ntoskrnl.

Reply:
https://git.reactos.org/?p=reactos.git;a=blob;f=ntoskrnl/config/cmdata.c;hb=04e92516123dd094bcfa3fd14b4d0d59c8e4eaf0#l68

Reply:
Already discussed.